document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java

```diff
@@ -300,6 +300,22 @@ public void testSelectAll() throws IOException {
     assertEquals(1, collection.count());
   }
 
+  @Test
+  public void testDeleteByFilter() throws IOException {
```

Comment:
Also, please add a test to validate exception is thrown without a filter.

Same with Postgres.